packages/datadog-esbuild/index.js

```diff
@@ -245,29 +244,33 @@ ${build.initialOptions.banner.js}`
         }
       }
 
-      const packageJson = JSON.parse(fs.readFileSync(pathToPackageJson).toString())
+      try {
+        const packageJson = JSON.parse(fs.readFileSync(pathToPackageJson).toString())
 
-      const isESM = isESMFile(fullPathToModule, pathToPackageJson, packageJson)
-      if (isESM && !interceptedESMModules.has(fullPathToModule)) {
-        fullPathToModule += ESM_INTERCEPTED_SUFFIX
-      }
+        const isESM = isESMFile(fullPathToModule, pathToPackageJson, packageJson)
+        if (isESM && !interceptedESMModules.has(fullPathToModule)) {
+          fullPathToModule += ESM_INTERCEPTED_SUFFIX
+        }
 
-      if (DEBUG) console.log(`RESOLVE: ${args.path}@${packageJson.version}`)
+        if (DEBUG) console.log(`RESOLVE: ${args.path}@${packageJson.version}`)
 
-      // https://esbuild.github.io/plugins/#on-resolve-arguments
-      return {
-        path: fullPathToModule,
-        pluginData: {
-          version: packageJson.version,
-          pkg: extracted.pkg,
-          path: extracted.path,
-          full: fullPathToModule,
-          raw: args.path,
-          pkgOfInterest: true,
-          kind: args.kind,
-          internal,
-          isESM
+        // https://esbuild.github.io/plugins/#on-resolve-arguments
+        return {
+          path: fullPathToModule,
+          pluginData: {
+            version: packageJson.version,
+            pkg: extracted.pkg,
+            path: extracted.path,
+            full: fullPathToModule,
+            raw: args.path,
+            pkgOfInterest: true,
+            kind: args.kind,
+            internal,
+            isESM
+          }
         }
+      } catch {
+        // Skip vendored dependencies which never have a `package.json`.
```

Comment:
This looks confusing for me. I think we should not fail. I understand that the package.json is not found of the dependency, we just still have to be sure the code will be bundled.